iterator.go

```diff
@@ -151,23 +152,35 @@ func (it *Iterator) Value() []byte {
 	return ret
 }
 
-func (it *Iterator) AllForPrefixes(prefixes [][]byte) (map[string][][]byte, error) {
+func (it *Iterator) AllForPrefixes(prefixes [][]byte, limit *int32, lastKey []byte) (map[string][][]byte, error) {
 	res := make(map[string][][]byte)
-
+	values := int32(0)
 	var err error
 
+	if (lastKey != nil) {
+		if _, err := it.Seek(lastKey); err != nil {
+			return nil, err
+		}
+	}
+
 	for _, prefix := range prefixes {
 		ok := false
-		if ok, err = it.Seek(prefix); err != nil {
-			return nil, err
+
+		if lastKey == nil || bytes.Compare(lastKey, prefix) <= 0  {
+			fmt.Printf("Seeking for %v\n", prefix)
+			if ok, err = it.Seek(prefix); err != nil {
+				return nil, err
+			}
+		} else {
+			ok = true
 		}
 
 		acc := make([][]byte, 0, 1)
 
-		for ok && bytes.HasPrefix(it.key, prefix) {
+		for ok && bytes.HasPrefix(it.key, prefix) && (limit == nil || values < *limit){
```

Comment:
i think we could just make limit an `int32` and assign it to a default value if it is `<=0` ?